Quantum_Basics.jl

```diff
@@ -0,0 +1,1888 @@
+md"In order to gain further intuition for the behaviour of states as the time evolution is applied, we turn to visualization. Since the full state including complex phases depends on 4 numbers it is difficult to imagine for us. However, a global complex phase, which is applied to the entire state $e^{i\phi} \Psi$, does not change the outcomes of measurements, so we can ignore it. All that remains is the relative complex phase between the entries, and the real magnitude of each entry of the vector.
+
+Show that a global complex phase does not change the outcome of measurements!"
+
+# ╔═╡ 592fafe8-ec65-486c-a541-f38da30d2dd2
+if probability(eigvecs(σ_x)[:,2], [0,1]) == probability(eigvecs(σ_x)[:,2]*cis(3), [0,1])
+	print("Global phases don't matter")
+else
+	print("Global phases do matter")
+end
+
+# ╔═╡ 460c778d-969c-4234-9e65-ef8f1bb3557f
+md"The outcome of a measurement is only dependent on the overlap of the eigenstates with the current state. This overlap is not changed by a global complex phase.
+
+We can further reduce the dimensionality of the problem by noting that the normalisation to length 1 of the vector removes one more degree of freedom. Therefore, only two numbers are sufficient to characterize the state: 
+ - The complex phase between the states
+ - How much the state points in the z-direction"
+
+# ╔═╡ f6b42b7d-c614-4c95-a105-d7dced988825
+function parametrize_Bloch_sphere(state)
+	return [2*acos(abs(state[1])),angle(state[1]) - angle(state[2])]
+end
+
+# ╔═╡ f802e2f0-201e-490b-937c-82e25bb640e0
+function Bloch_point(state)
+	parameters = parametrize_Bloch_sphere(state)
+	return Float64[sin(parameters[1])*cos(parameters[2]), sin(parameters[1])*sin(parameters[2]), cos(parameters[1])]
+end
+
+# ╔═╡ 70539c8d-7565-4554-b60e-22220151b38b
+time_range = 0:0.1:10
+
+# ╔═╡ f0716eed-8be2-4031-9345-c28386fc1412
+initial_state = [1,0]
+
+# ╔═╡ 4a5aa91c-cd22-481c-9e99-bf24a1eaa092
+@bind time PlutoUI.Slider(time_range)
+
+# ╔═╡ a0c4484b-0869-427b-9885-de3072bc8fb4
+md"Because the point can be a bit difficult to follow, we also plot the trace. Write the name of the function, which returns the time-evolution operator here:"
+
+# ╔═╡ a6269134-0ca9-464a-b05e-e4a17c1eac46
+time_evol = time_evolution_B_x
+
+# ╔═╡ 67988f84-dc78-4c9d-bc6b-260fd36556a6
+time_evolved_state = time_evol(time)*initial_state
+
+# ╔═╡ 28728934-b4b5-4b58-b44a-2ea9c5f51f5b
+plotted_point = Bloch_point(time_evolved_state)
+
+# ╔═╡ ca85162f-4b9a-4d12-89f1-2126321e4071
+let # Define the sphere's radius and parameterize the surface using spherical coordinates
+radius = 1
+u = range(0, stop=2π, length=50)
+v = range(0, stop=π, length=50)
+x = [radius * sin(v[j]) * cos(u[i]) for i in 1:length(u), j in 1:length(v)]
+y = [radius * sin(v[j]) * sin(u[i]) for i in 1:length(u), j in 1:length(v)]
+z = [radius * cos(v[j]) for i in 1:length(u), j in 1:length(v)]
+
+plotted_points = zeros(Float64, (3,length(time_range)))
+
+for i in 1:length(time_range)
```

Comment:
Maybe split this up into multiple cells? I'm not sure, but I find multiple cells often makes code look cleaner, even if it is longer. The exception for me is plotting code.

A more elegant solution might be to use `zip` and `eachcol`. So, 

```Julia
pnts = zeros(Float64, (3, times))
for (t, col) in zip(times, eachcol(pnts))
    col .= to_bloch(propagator(t) * initial_state)
end
```

I've renamed a few things to be a bit shorter. Maybe a function `propagate(time, state)` would be good, too. This way you can avoid allocations. You could also use StaticArrays. Performance doesn't matter at this stage, but it's good to get in the habit of avoiding allocations when you don't need to allocate.